app/scripts/controllers/permissions/background-api.js

```diff
@@ -147,20 +185,56 @@ export function getPermissionBackgroundApiMethods({
     addPermittedAccounts: (origin, accounts) =>
       addMoreAccounts(origin, accounts),
 
-    removePermittedAccount: (origin, account) => {
+    removePermittedAccount: (origin, address) => {
       const caip25Caveat = getCaip25Caveat(origin);
       if (!caip25Caveat) {
         throw new Error(
-          `Cannot remove account "${account}": No permissions exist for origin "${origin}".`,
+          `Cannot remove account "${address}": No permissions exist for origin "${origin}".`,
         );
       }
 
-      const existingAccounts = getEthAccounts(caip25Caveat.value);
-
-      const remainingAccounts = existingAccounts.filter(
-        (existingAccount) => existingAccount !== account,
+      const permittedAccounts = new Set();
+      Object.values(caip25Caveat.value.requiredScopes).forEach(
+        ({ accounts }) => {
+          accounts.forEach((account) => {
+            permittedAccounts.add(account);
+          });
+        },
+      );
+      Object.values(caip25Caveat.value.optionalScopes).forEach(
+        ({ accounts }) => {
+          accounts.forEach((account) => {
+            permittedAccounts.add(account);
+          });
+        },
       );
 
+      const existingAccounts = Array.from(permittedAccounts);
```

Comment:
this logic needs to be DRY'ed into a new getter in core, but again, let's not block on this merging into sip-26 branch

Reply:
Should we add a todo comment inline to remind us?